test/common/stats/allocator_impl_test.cc

```diff
@@ -75,6 +77,30 @@ TEST_F(AllocatorImplTest, GaugesWithSameName) {
   EXPECT_EQ(0, g2->value());
 }
 
+TEST_F(AllocatorImplTest, Threads) {
```

Comment:
Can you add a comment on what this test tests with a link to the issue/explanation/etc.? IMO it would be good to also use my new synchronizer stuff to have an explicit test for this case?

Reply:
done, but IDK if it is worth it. The test is complicated and (when it passes) it needs a timeout to complete because having bug fixed makes the threads deadlock each other: one holds the lock and is blocked on the synchronizer, the other one is waiting for that lock.